src/main/java/org/folio/consortia/messaging/listener/ConsortiaEventListener.java

```diff
@@ -32,9 +29,9 @@ public class ConsortiaEventListener {
     concurrency = "#{folioKafkaProperties.listener['user-created'].concurrency}",
     containerFactory = "kafkaListenerContainerFactory")
   public void userCreatedListener(String data, MessageHeaders messageHeaders) {
-    String centralTenantId = getCentralTenantId(messageHeaders); // to create affiliation in central tenant schema
-    runInFolioContext(createFolioExecutionContext(messageHeaders, moduleMetadata, centralTenantId),
-      () -> userAffiliationService.createPrimaryUserAffiliation(data));
+    // to create affiliation in central tenant schema
+    String centralTenantId = configurationService.getCentralTenantByIdByHeader(messageHeaders);
+    runInFolioContext(createFolioExecutionContext(messageHeaders, moduleMetadata, centralTenantId), () -> userAffiliationService.createPrimaryUserAffiliation(data));
```

Comment:
I think better to remain context switching in one place, I propose to remain it in the listener layer,
because if context swithcing in 2 different places for the same operation - it is hard to understand the flow.

Here we need 2 context switches:

- we receiving kafka event for university tenant - > switch to university context -> call university DB schema to find central tenant

- we now what is central tenant id and should swich context to the central tenant to save all data in the central DB schema

Reply:
Ok, I will move to listener class